test/functional/test_runner.py

```diff
@@ -364,9 +364,10 @@ def main():
 def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, enable_coverage=False, args=None, combined_logs_len=0, failfast=False, runs_ci, use_term_control):
     args = args or []
 
-    # Warn if bitcoind is already running (unix only)
+    # Warn if bitcoind is already running
+    # On macOS pidof (installed via brew) returns b''
     try:
-        if subprocess.check_output(["pidof", "bitcoind"]) is not None:
+        if subprocess.check_output(["pidof", "bitcoind"]) not in [None, b'']:
```

Comment:
I think the `None` can be removed. `check_output` should either fail or return an (empty) string.